File: src/examples/kotlin/data-classes.kt
export default {
  id: 'data-classes-kotlin',
  language: 'kotlin',
  title: 'Kotlin Data Classes',
  description: 'Data classes with destructuring, copy, and component functions',
  difficulty: 'beginner',
  category: 'oop',
  tags: ['data-classes', 'kotlin', 'oop'],
  code: `data class Person(
  val name: String,
  val age: Int,
  val email: String = ""
)

fun main() {
  val alice = Person("Alice", 30, "alice@example.com")
  val bob = alice.copy(name = "Bob", email = "bob@example.com")

  // Destructuring
  val (name, age, email) = alice
  println("$name is $age years old")

  println("Alice: $alice")
  println("Bob:   $bob")
  println("Same? ${alice == bob}")
}`
};

File: src/examples/javascript/data-classes.js
/**
 * Data Classes / Records - JavaScript
 *
 * Working with structured data objects, object methods, and immutable patterns.
 */
export default {
  id: 'data-classes-js',
  language: 'javascript',
  title: 'Data Classes & Objects',
  description: 'Structured data objects, factory functions, and immutability',
  difficulty: 'beginner',
  category: 'oop',
  tags: ['objects', 'data', 'patterns', 'OOP'],
  code: `// Factory function for creating structured data objects
function createProduct(id, name, price, category) {
  return Object.freeze({ id, name, price, category });
}

// Create some products
const laptop = createProduct(1, "MacBook Pro", 1999.99, "electronics");
const book = createProduct(2, "Clean Code", 35.99, "books");
const coffee = createProduct(3, "Ethiopia Blend", 18.50, "food");

const inventory = [laptop, book, coffee];

// Computed properties
const total = inventory.reduce((sum, p) => sum + p.price, 0);
const byCategory = inventory.reduce((acc, p) => {
  acc[p.category] = acc[p.category] || [];
  acc[p.category].push(p.name);
  return acc;
}, {});

console.log("Total value: $" + total.toFixed(2));
console.log("Categories:", byCategory);

// Updating immutable objects (spread operator)
const discounted = { ...laptop, price: laptop.price * 0.9 };
console.log("Original price: $" + laptop.price);
console.log("Discounted: $" + discounted.price.toFixed(2));`,
};
